net.go

```diff
@@ -389,27 +389,16 @@ func collectInstanceQueryOptions(opts []InstanceQueryOption) (instanceQueryOptio
 }
 
 // GetInstancesByVIPAddress returns the set of instances registered with the given VIP address,
-// potentially filtered per the constraints supplied as options.
+// selecting either an insecure or secure VIP address with the given name, potentially filtered
+// per the constraints supplied as options.
 //
 // NB: The VIP address is case-sensitive, and must match the address used at registration time.
-func (e *EurekaConnection) GetInstancesByVIPAddress(addr string, opts ...InstanceQueryOption) ([]*Instance, error) {
+func (e *EurekaConnection) GetInstancesByVIPAddress(addr string, secure bool, opts ...InstanceQueryOption) ([]*Instance, error) {
 	options, err := collectInstanceQueryOptions(opts)
 	if err != nil {
 		return nil, err
 	}
-	return e.getInstancesByVIPAddress(addr, insecure, options)
-}
-
-// GetInstancesBySecureVIPAddress returns the set of instances registered with the given secure
-// VIP address, potentially filtered per the constraints supplied as options.
-//
-// NB: The secure VIP address is case-sensitive, and must match the address used at registration time.
-func (e *EurekaConnection) GetInstancesBySecureVIPAddress(addr string, opts ...InstanceQueryOption) ([]*Instance, error) {
-	options, err := collectInstanceQueryOptions(opts)
-	if err != nil {
-		return nil, err
-	}
-	return e.getInstancesByVIPAddress(addr, secure, options)
+	return e.getInstancesByVIPAddress(addr, vipAddressKind(secure), options)
```

Comment:
I would drop `vipAddressKind` all together. It's used only in one place and it actually obfuscate the meaning in my opinion (kind instead of secure).